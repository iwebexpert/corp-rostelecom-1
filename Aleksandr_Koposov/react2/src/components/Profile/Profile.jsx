import React, { useEffect, useState } from 'react'

import {
  Card,
  CardContent,
  CardActions,
  Badge,
  Icon,
  Avatar,
  TextField,
  Button,
  Dialog,
  DialogTitle,
  CircularProgress
} from '@material-ui/core'

import './Profile.scss'

export const Profile = (props) => {
  const { user, isLoading } = props
  const { id, name, age, about, avatar } = user || {}
  const [profileState, setProfile] = useState(
    {
      init: false,
      dialog: false,
      id,
      name,
      age,
      about,
      avatar
    }
  )

  useEffect(() => {
    setProfile({
      ...profileState,
      init: true,
      ...props.user
    })
  }, [])

  const onInputChange = (event) => {
    setProfile({
      ...profileState,
      [event.target.name]: event.target.value
    })
  }

  const saveChanges = () => {
    const { onSave } = props
    const { id, name, age, about, avatar } = profileState
    onSave({ id, name, age, about, avatar })
  }

  const toggleDialog = () => {
    setProfile({
      ...profileState,
      dialog: !profileState.dialog
    })
  }

  const avatarSize = {
    width: 128,
    height: 128
  }

  return (
    <Card className="profile" elevation={3}>
      <CardContent>
        <Badge
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          badgeContent={<Icon>edit</Icon>}
          onClick={toggleDialog}
        >
          {!isLoading ? (
            <Avatar
              style={avatarSize}
              alt="Аватар пользователя"
              src={profileState.avatar}
            />
          ) : (
              <Avatar style={avatarSize}>
                <CircularProgress />
              </Avatar>
            )}
        </Badge>
        <Dialog
          onClose={toggleDialog}
          open={profileState.dialog}
          className="avatar-dialog"
        >
          <DialogTitle>Выбор аватара</DialogTitle>
          <Card>
            <CardContent>
              <TextField
                name="avatar"
                label="Ссылка на аватар (можете изменять число в конце для теста)"
                variant="outlined"
                value={profileState.avatar}
                onChange={onInputChange}
                style={{ width: '550px' }}
              />
            </CardContent>
          </Card>
        </Dialog>
        {!isLoading ?
          (
            <div>
              <TextField
                name="name"
                label="Имя"
                variant="outlined"
                value={profileState.name}
                onChange={onInputChange}
              />
              <TextField
                name="age"
                label="Возраст"
                variant="outlined"
                type="number"
                value={profileState.age}
                onChange={onInputChange}
              />
              <TextField
                name="about"
                label="О себе"
                multiline
                variant="outlined"
                value={profileState.about}
                onChange={onInputChange}
              />
            </div>
          ) : (
            <div style={{
              width: '100%',
              textAlign: 'center'
            }}>
              <CircularProgress />
            </div>
          )}
      </CardContent>
      <CardActions>
        <Button
          color="primary"
          onClick={saveChanges}
        >Сохранить</Button>
      </CardActions>
    </Card>
  )
}
