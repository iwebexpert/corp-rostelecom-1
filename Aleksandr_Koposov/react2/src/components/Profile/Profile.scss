.profile {
  & .MuiBadge-badge {
    height: 36px;
    width: 36px;
    border-radius: 50%;
    background-color: grey;
    color: white;
    cursor: pointer;
  }

  & .MuiTextField-root {
    width: 100%;
    margin: 15px 0;
  }
}

.avatar-dialog {
  & .MuiTextField-root {
    max-width: 550px;
    width: 100%;
  }
}
