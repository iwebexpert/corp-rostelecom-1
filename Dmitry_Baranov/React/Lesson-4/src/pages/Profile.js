import React from 'react';

export function ProfilePage(){
    return (
        <div>
            <h3>Profile page</h3>
            <p><i>Имя: </i>Имя пользователя</p>
            <p><i>Ник: </i>Псевдоним пользователя</p>
            <p><i>Возраст: </i>Возраст пользователя</p>
        </div>
    );
}