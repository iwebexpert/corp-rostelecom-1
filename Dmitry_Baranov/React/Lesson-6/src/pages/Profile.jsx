import React from 'react';
import {profile} from '../helpers/profilesData';

export function ProfilePage() {
    return (
        <div>
            <h3>Profile page</h3>
            <ul>
                <li>Имя: {profile[0].name}</li>
                <li>Возраст: {profile[0].age}</li>
                <li>E-mail: {profile[0].email}</li>
            </ul>
        </div>
    );
}