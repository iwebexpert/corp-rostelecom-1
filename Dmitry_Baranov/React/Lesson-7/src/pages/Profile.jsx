import React, { Component } from 'react';


export class Profile extends Component {
    render() {
        const profile = this.props.profile;
        console.log("profile:", profile);
        const { isLoading, isError } = this.props;

        if (isError) {
            return (<div>Не удалось загрузить профиль...</div>);
        }

        if (isLoading) {
            return (<div>Loading...</div>);
        }
        return (
            <div>Profile page
                <ul>
                    <li>
                        Имя: {profile.name}
                    </li>
                    <li>
                        Возраст: {profile.age}
                    </li>
                    <li>
                        e-mail: {profile.email}
                    </li>

                </ul>
            </div>
        );
    }

}