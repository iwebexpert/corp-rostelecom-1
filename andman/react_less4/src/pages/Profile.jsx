import React from 'react';

export function ProfilePage() {
    return (
        <div>
            <h2>Profile page</h2>
            E-mail:  test@mail.ru <hr />
            Логин: Test1 <hr />
        </div>


    );
}