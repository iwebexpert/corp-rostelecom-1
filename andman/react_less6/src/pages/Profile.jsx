import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {
        const { profile } = this.props;
        //console.log(profile);
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );
    };
}