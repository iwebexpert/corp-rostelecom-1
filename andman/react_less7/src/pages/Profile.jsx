import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {

        const { profile, isLoading, isError } = this.props;
        if (isError) {
            return (
                <div>Не удалось загрузить профиль...</div>
            );
        }
        if (isLoading) {
            return (
                <div>Profile loading...</div>
            );
        }
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );

    };
}