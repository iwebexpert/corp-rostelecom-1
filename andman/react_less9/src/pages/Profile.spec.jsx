import { shallow } from 'enzyme';
import React from 'react';
import { ProfilePage } from './Profile';

const profile = { name: 'WebDev', email: 'webdev@webdev.com', avatar: 'ava.jpg' };

describe('ProfilePage component', () => {
    it('should render ProfilePage component (1)', () => {
        const component = shallow(<ProfilePage
            profile={profile}
            isLoading={false}
            isError={false}
        />);
        expect(component).toMatchSnapshot();
    });

    it('should render ProfilePage component (loading)', () => {
        const component = shallow(<ProfilePage
            profile={profile}
            isLoading={true}
            isError={false}
        />);
        expect(component).toMatchSnapshot();
    });

    it('should render ProfilePage component (error)', () => {
        const component = shallow(<ProfilePage
            profile={profile}
            isLoading={false}
            isError={true}
        />);
        expect(component).toMatchSnapshot();
    });

    it('should render ProfilePage component (loading & error)', () => {
        const component = shallow(<ProfilePage
            profile={profile}
            isLoading={true}
            isError={true}
        />);
        expect(component).toMatchSnapshot();
    });
});