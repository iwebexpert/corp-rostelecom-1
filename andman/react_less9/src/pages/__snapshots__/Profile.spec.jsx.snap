// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component should render ProfilePage component (1) 1`] = `
<div>
  <h2>
    Profile page
  </h2>
  E - mail: 
  webdev@webdev.com
   
  <hr />
  Логин: 
  WebDev
   
  <hr />
  <img
    height="200px"
    src="ava.jpg"
    width="200px"
  />
</div>
`;

exports[`ProfilePage component should render ProfilePage component (error) 1`] = `
<div>
  Не удалось загрузить профиль...
</div>
`;

exports[`ProfilePage component should render ProfilePage component (loading & error) 1`] = `
<div>
  Не удалось загрузить профиль...
</div>
`;

exports[`ProfilePage component should render ProfilePage component (loading) 1`] = `
<div>
  Profile loading...
</div>
`;
